webclient lose the session when session.forget is used

When `session.forget` is added to an action like `index` we can observe that on subsequent call a logged user is logged out.

To reproduce, modify  `TestWeb.testRegisterAndLogin` in gluon/tests/test_web.py (tryed on  commit e6a3081b42ecd58441419930266baf286561c4c7) so as adding `session.forget(response)` at the first line of the `index` controller.

The test is run with :

```
python2 -m unittest -v gluon.tests.test_web.TestWeb.testRegisterAndLogin
```

And fails with :

```
======================================================================
FAIL: testRegisterAndLogin (gluon.tests.test_web.TestWeb)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "gluon/tests/test_web.py", line 92, in testRegisterAndLogin
    self.assertTrue('Welcome Homer' in client.text)
AssertionError: False is not true

----------------------------------------------------------------------
```

Which seems unexpected according to the google group exchanges about this subject.

I'd like to help, but I have only a rough idea of the expected behaviour of a browser with respect to cookies, sessions and headers… Link to documentation is welcomed.

--- a/gluon/contrib/webclient.py
+++ b/gluon/contrib/webclient.py
@@ -43,6 +43,7 @@ class WebClient(object):
         self.forms = {}
         self.history = []
         self.cookies = {}
+        self.cookiejar = cookielib.CookieJar()
         self.default_headers = default_headers
         self.sessions = {}
         self.session_regex = session_regex and re.compile(session_regex)
@@ -79,9 +80,8 @@ class WebClient(object):
         cookies = cookies or {}
         headers = headers or {}
 
-        cj = cookielib.CookieJar()
         args = [
-            urllib2.HTTPCookieProcessor(cj),
+            urllib2.HTTPCookieProcessor(self.cookiejar),
             urllib2.HTTPHandler(debuglevel=0)
             ]
         # if required do basic auth

--- a/gluon/tests/test_web.py
+++ b/gluon/tests/test_web.py
@@ -110,7 +110,7 @@ class TestWeb(LiveTest):
         self.assertTrue('Welcome Homer' in client.text)
 
         client = WebClient('http://127.0.0.1:8000/admin/default/')
-        client.post('index', data=dict(password='hello'))
+        client.post('index', data=dict(password='testpass'))
         client.get('site')
         client.get('design/welcome')
 

